### PR TITLE
Caffe2Fluid:Support custom layer conversion

### PR DESCRIPTION
**Pull request by walloollaw**
In order to support custom layer conversion from caffe to fluid,  we need to upgrade this tool.
new features for this upgration:
1, support custom caffe layer conversion which enables users to convert their own caffe layer
2, add several examples to show how to extend this tool to convert a custom layer
3, more info about custom layer is provided in README.md

### CODE REVIEW

File: fluid/image_classification/caffe2fluid/README.md [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
-### Howto
+### HowTo
 1. Prepare caffepb.py in ./proto if your python has no 'pycaffe' module, two options provided here:
 - Generate pycaffe from caffe.proto
   <pre><code>bash ./proto/compile.sh</code></pre>
 
 - download one from github directly
-  <pre><code>cd proto/ && wget https://github.com/ethereon/caffe-tensorflow/blob/master/kaffe/caffe/caffepb.py
-</code></pre>
+  <pre><code>cd proto/ && wget https://github.com/ethereon/caffe-tensorflow/blob/master/kaffe/caffe/caffepb.py</code></pre>
```
Comment (qingqing01) [inline fluid/image_classification/caffe2fluid/README.md]: Do not use `<pre><code>  </code></pre>` 

Use code syntax:

https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting

File: fluid/image_classification/caffe2fluid/README.md [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
         --caffemodel alexnet.caffemodel \
         --data-output-path alexnet.npy \
-        --code-output-path alexnet.py
-</code></pre>
+        --code-output-path alexnet.py</code></pre>
 
 - save weights as fluid model file
-  <pre><code>python alexnet.py alexnet.npy ./fluid_model
-</code></pre>
+  <pre><code>python alexnet.py alexnet.npy ./fluid_model</code></pre>
```
Comment (qingqing01) [inline fluid/image_classification/caffe2fluid/README.md]: All these code styles are needed to be fixed as above

File: fluid/image_classification/caffe2fluid/README.md [hunk 3]
```diff
@@ -1,33 +1,49 @@
...
 
 4. compare the inference results with caffe
 - see more details in '*examples/imagenet/diff.sh*'
 
+### How to convert custom layer
+1. implement your custom layer in a file under '*kaffe/custom_layers*', eg: mylayer.py
+- implement a 'shape_func(input_shape, [other_caffe_params])' to calculate the output shape
+- implement a 'layer_func(inputs, name, [other_caffe_params])' to construct a fluid layer
+- register these two functions using '*register*'
+- notes: more examples can be found in '*kaffe/custom_layers*'
```
Comment (qingqing01) [inline fluid/image_classification/caffe2fluid/README.md]: 1. 首字母大写。
2. 所有-需要加缩进。
3. 注意代码格式 不是''  是 ``

File: fluid/image_classification/caffe2fluid/kaffe/custom_layers/__init__.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
+"""
+"""
+
+from .register import get_registered_layers
+#custom layer import begins
```
Comment (qingqing01) [inline fluid/image_classification/caffe2fluid/kaffe/custom_layers/__init__.py:5]: Remove the unused code.

Comment (walloollaw) [reply fluid/image_classification/caffe2fluid/kaffe/custom_layers/__init__.py:5]: this line is marked for the purpose of adding new layers implemented by users

**Comment by walloollaw** [comment]
sorry to make these commits dirty, so i created another PR here [PR-866](https://github.com/PaddlePaddle/models/pull/866)

**Pull request closed without merging**